### PR TITLE
docs(ref): Specify 'rust_version' in Index format

### PR DESCRIPTION
**Pull request by epage**
See rust-lang/crates.io#6267

### CODE REVIEW

File: src/doc/src/reference/registry-index.md [hunk 1]
```diff
@@ -199,6 +199,9 @@ explaining the format of the entry.
     "features2": {
         "serde": ["dep:serde", "chrono?/serde"]
     }
+    // Required version of Rust (optional).
```
Comment (Turbo87) [inline src/doc/src/reference/registry-index.md:202]:
```suggestion
    // Minimum required version of Rust (optional).
```

Comment (weihanglo) [reply src/doc/src/reference/registry-index.md:202]: Or be consistent with the current doc — The minimal supported Rust version.

https://doc.rust-lang.org/nightly/cargo/reference/manifest.htm

**Review by weihanglo — COMMENTED** [review]
Cross linked to https://github.com/rust-lang/crates.io/pull/6267

File: src/doc/src/reference/registry-index.md (hunk 1 above)
Comment (Turbo87) [reply src/doc/src/reference/registry-index.md:202]: yeah, that sounds even better :)

**Comment by Eh2406** [comment]
Do we want to mention that is not currently available from the publish JSON?
Do we want to mention that it is currently not read by cargo and is optional?
Both of which may change in the future.

**Comment by weihanglo** [comment]
crates.io 503: <https://github.com/rust-lang/cargo/actions/runs/4810738015/jobs/8563779797?pr=12040>
deployment history: <https://rust-lang.zulipchat.com/#narrow/stream/318791-t-crates-io/topic/deployments/near/352857644>

cc <https://github.com/rust-lang/crates.io/issues/6385>

**Comment by epage** [comment]
> Do we want to mention that is not currently available from the publish JSON?

As there is a section to call out differences between the Publish API and the Index, I've noted it there

> Do we want to mention that it is currently not read by cargo and is optional?

I do mention it is optional.  We don't have a location to point people to yet and cargo's use of it which makes me hesitant to do so.

**Comment by Turbo87** [comment]
yeah, same thing as I mentioned yesterday, during deploys of crates.io itself apparently we are getting more and more 503s these days :-/

**Review by weihanglo — APPROVED** [review]

**Comment by weihanglo** [comment]
Thanks for the clarification.

@bors r+

**Pull request merged**